contracts/Identity.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity ^0.4.25;
+
+import "./lib/ECDSA.sol";
+
+
+contract Identity {
+
+    address private owner;
+    bool private initialised;
+
+    mapping(bytes32 => bool) private hashUsed;
+
+    event TransactionExecution(bytes32 hash, bool status);
+
+    constructor() public {
+        // don't do anything here to allow usage of proxy contracts.
+    }
+
+    // This contract can receive ether
+    function () public payable {}
+
+    function init(address _owner) public {
+        require(! initialised, "The contract has already been initialised.");
+        owner = _owner;
+        initialised = true;
+    }
+
+    function executeTransaction(
+        address from,
+        address to,
+        uint256 value,
+        bytes data,
+        uint256 nonce,
+        bytes extraHash,
```

Comment:
Why would you pass an extraHash here. I think you want to pass `extraData` and hash this data to make `extraHash` that you then add to the signed `hash` representing the transaction.

Reply:
This should be extra_data yes. We dont want to hash it though, so that it can serve it s purpose for backwards compatibility.